saleor/graphql/checkout/resolvers.py

```diff
@@ -34,30 +39,28 @@ def resolve_checkout(info, token, id):
         .filter(token=token)
         .first()
     )
-
     if checkout is None:
         return None
-
-    # resolve checkout in active channel
+    # always return checkout for active channel
     if checkout.channel.is_active:
-        # resolve checkout for anonymous customer
-        if not checkout.user:
-            return checkout
-
-        # resolve checkout for logged-in customer
-        user = info.context.user
-        if user and checkout.user == user:
-            return checkout
+        return checkout
 
-    # resolve checkout for staff user
+    # resolve checkout for staff or app
     requester = get_user_or_app_from_context(info.context)
 
     if not requester:
         return None
```

Comment:
Why `None`? You already changed the way in line 63/60 to raise PermissionDenied.

Reply:
should we raise the same error or more specyfic for a case where the requestor is not found?

Reply:
When requestor is not found, then it means that the request is anonymous, and try to fetch the details for the in-active channel (at least from what I understand 😅 )

Reply:
ok so throwing an PermissionDenied should be good option here